packages/wast-parser/src/tokenizer.js

```diff
@@ -279,62 +331,14 @@ function tokenize(input: string) {
       char === "-" ||
       char === "+"
     ) {
-      let value = "";
-      const startColumn = column;
+      const value = numberLiteralFSM.run(input.slice(current));
```

Comment:
Above we test if it's a literal before going through the FSM. Is there a way to handle errors into the FSM so we can remove the pre-condition here?

Reply:
Hmm I have to think about this, not sure. We could run the fsm at this point and check whether the output is an empty string or not. I don't think that will behave completely the same currently, but it should be doable.

I prefer this solution, because I think running the fsm is expensive (currently copying complete remaining string) so it should only be ran when we know theres a number literal. I'm not completely sure about the perf aspect though

Reply:
I think we can add an error state, but the perf concern is valid. We can leave it as is for now :+1: